Makefile

```diff
@@ -0,0 +1,51 @@
+# Variables
+GENERATOR := openapi-generator 
+OUTPUT_DIR := .
+OPENAPI_FILE := api/openapi.yaml
+YQ := ./yq  # Use the local yq binary
+
+# Install dependencies
+install-deps:
+	@if [ ! -x $(YQ) ]; then \
+		echo "Installing yq..."; \
+		curl -L https://github.com/mikefarah/yq/releases/latest/download/yq_darwin_amd64 -o $(YQ); \
+		chmod +x $(YQ); \
+		echo "yq installed locally as $(YQ)."; \
+	else \
+		echo "yq is already installed."; \
+	fi
+
+# Install OpenAPI Generator CLI
+install-generator:
+	@if ! command -v $(GENERATOR) &> /dev/null; then \
+		echo "Installing openapi-generator-cli via Homebrew..."; \
+		brew install openapi-generator; \
+	else \
+		echo "openapi-generator-cli is already installed."; \
+	fi
+
+# Show modifiable paths
+show-paths:
+	@if [ ! -x $(YQ) ]; then \
```

Comment:
This `if` wouldn't be needed because the `yq` alias would automatically pull the Docker image if it's not available on the host.